cmd/secrets-store-csi-driver/main.go

```diff
@@ -46,6 +49,8 @@ var (
 	// grpcSupportedProviders is a ; separated string that can contain a list of providers. The reason it's a string is to allow scenarios
 	// where the driver is being used with 2 providers, one which supports grpc and other using binary for provider.
 	grpcSupportedProviders = flag.String("grpc-supported-providers", "", "set list of providers that support grpc for driver-provider [alpha]")
+	enableSecretRotation   = flag.Bool("enable-secret-rotation", false, "Enable secret rotation feature [alpha]")
+	rotationPollInterval   = flag.Duration("rotation-poll-interval", 60, "Secret rotation poll interval duration")
```

Comment:
💯 on the SPC. The lifecycle of a secret is configurable in Vault, so secrets would have different rotation periods.  This would be a nice to have but not absolutely required since we could add to the Vault Agent sidecar to the secret store pod with caching enabled. The agent would renew secrets, know if secrets are still valid or serve new secrets if they're expired. 

We could set a very aggressive poll interval and lean on the sidecar to make it work with Vault.

Reply:
> It might be nice for this to be a configurable option in the SPC CRD instead of a flag on the controller, but this seems good to start.

@tam7t What is the use case for this? Is the idea that secrets have different lifetime period? Having a global poll interval flag makes it consistent across all the SPCs. The provider would get the current version and the latest version. Only if latest != current would a write to the fs be required.

Reply:
I've opened an issue for this - https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/322